keras_core/layers/preprocessing/hashed_crossing_test.py

```diff
@@ -71,3 +72,104 @@ def test_tf_data_compatibility(self):
         for output in ds.take(1):
             output = output.numpy()
         self.assertAllClose(np.array([1, 4, 1, 1, 3]), output)
+
+    def test_upsupported_shape_input_fails(self):
+        with self.assertRaisesRegex(ValueError, "inputs should have shape"):
+            layers.HashedCrossing(num_bins=10)(
+                (tf.constant([[[1.0]]]), tf.constant([[[1.0]]]))
+            )
+
+    def test_cross_output_dtype(self):
+        input_1, input_2 = np.array([1]), np.array([1])
+
+        layer = layers.HashedCrossing(num_bins=2)
+        output_dtype = backend.standardize_dtype(
+            layer((input_1, input_2)).dtype
+        )
+        self.assertEqual(output_dtype, "int64")
+        layer = layers.HashedCrossing(num_bins=2, dtype=tf.int32)
+        output_dtype = backend.standardize_dtype(
+            layer((input_1, input_2)).dtype
+        )
+        self.assertEqual(output_dtype, "int32")
+        layer = layers.HashedCrossing(num_bins=2, output_mode="one_hot")
+        output_dtype = backend.standardize_dtype(
+            layer((input_1, input_2)).dtype
+        )
+        self.assertEqual(output_dtype, "float32")
+        layer = layers.HashedCrossing(
+            num_bins=2, output_mode="one_hot", dtype=tf.float64
+        )
+        output_dtype = backend.standardize_dtype(
+            layer((input_1, input_2)).dtype
+        )
+        self.assertEqual(output_dtype, "float64")
+
+    def test_non_list_input_fails(self):
+        with self.assertRaisesRegex(ValueError, "should be called on a list"):
+            layers.HashedCrossing(num_bins=10)(tf.constant(1))
+
+    def test_single_input_fails(self):
+        with self.assertRaisesRegex(ValueError, "at least two inputs"):
+            layers.HashedCrossing(num_bins=10)([tf.constant(1)])
+
+    def test_sparse_input_fails(self):
+        with self.assertRaisesRegex(
+            ValueError, "inputs should be dense tensors"
+        ):
+            sparse_in = tf.sparse.from_dense(tf.constant([1]))
+            layers.HashedCrossing(num_bins=10)((sparse_in, sparse_in))
+
+    def test_float_input_fails(self):
```

Comment:
This uses tf.constant -- switch to `ops`

Reply:
Done